photon-api/src/main/scala/com/linkedin/photon/ml/data/RandomEffectDataSet.scala

```diff
@@ -488,21 +496,63 @@ object RandomEffectDataSet {
    */
   private def featureSelectionOnActiveData(
       activeData: RDD[(REId, LocalDataSet)],
-      randomEffectDataConfiguration: RandomEffectDataConfiguration): RDD[(REId, LocalDataSet)] = {
+      randomEffectDataConfiguration: RandomEffectDataConfiguration,
+      featureShardStatsOpt: FeatureShardStatisticsOpt,
+      globalPositiveInstance: Array[Double]): RDD[(REId, LocalDataSet)] = {
+
+
+    featureShardStatsOpt match {
+      case Some(featureShardStats) => {
+        val featureShardId = randomEffectDataConfiguration.featureShardId
+        val (binaryIndices, nonBinaryIndices) = filterBinaryFeatures(featureShardStats(featureShardId))
```

Comment:
"Filter" isn't a good name for this, because it's not really filtering a collection. When you see `filter`, you expect it to be (seq in, seq out).